core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -229,13 +231,16 @@ private synchronized void notifyTerminated() {
   private class InProcessStream {
     private final InProcessServerStream serverStream = new InProcessServerStream();
     private final InProcessClientStream clientStream = new InProcessClientStream();
+    private final StatsTraceContext serverStatsTraceContext;
     private final Metadata headers;
     private MethodDescriptor<?, ?> method;
 
     private InProcessStream(MethodDescriptor<?, ?> method, Metadata headers) {
       this.method = checkNotNull(method, "method");
       this.headers = checkNotNull(headers, "headers");
-
+      this.serverStatsTraceContext = checkNotNull(
+          serverTransportListener.methodDetermined(method.getFullMethodName(), headers),
```

Comment:
nit: I think I'd prefer `methodDetermined()` to be called in `newStream()` and the constructor just receives `clientStatsTraceContext` and `serverStatsTraceContext`. (Whether `clientStatsTraceContext` actually is passed in today doesn't matter much, but you can see how it would be plumbed from `newStream()`)